Emby.Server.Implementations/ApplicationHost.cs

```diff
@@ -769,20 +758,8 @@ protected async Task RegisterResources(IServiceCollection serviceCollection, ICo
             CertificateInfo = GetCertificateInfo(true);
             Certificate = GetCertificate(CertificateInfo);
 
-            HttpServer = new HttpListenerHost(
-                this,
-                LoggerFactory.CreateLogger<HttpListenerHost>(),
-                ServerConfigurationManager,
-                startupConfig,
-                NetworkManager,
-                JsonSerializer,
-                XmlSerializer,
-                CreateHttpListener())
-            {
-                GlobalResponse = LocalizationManager.GetLocalizedString("StartupEmbyServerIsLoading")
-            };
-
-            serviceCollection.AddSingleton(HttpServer);
+            serviceCollection.AddSingleton<IHttpListener, WebSocketSharpListener>();
+            serviceCollection.AddSingleton<IHttpServer, HttpListenerHost>();
```

Comment:
Construct `HttpListenerHost` and `WebSocketSharpListener` using the DI framework instead of manually. We should really be doing this for all the services/resources, but I have just updated this one so I could inject `IConfiguration` into `HttpListenerHost` correctly.

Note that I've added a new method to actually create this service [here](https://github.com/jellyfin/jellyfin/pull/2601/files#diff-5ff7dd3f2f09dc8b526fc45b5264635fR873)